### PR TITLE
convert_client: non-matching IP versions

### PR DESCRIPTION
**Pull request by GregoryVds**
This patch adds support in convert_client for non-matching IP versions
on upstream and downstream TCP connections.

Up to now, the convert_client was making the incorrect assumption that
the client would only reach IPv4 hosts when using a Converter addressable
over IPv4 (and similarly for IPv6). However, it should be possible to
reach both an IPv4 and IPv6 end server through a Converter addressable
over IPv4 (same goes for IPv6). IP versions used on upstream and
downstream don't have to match.

In particular, this was posing serious problems when using the toy
convert_client with an end-server such as "google.com" which resolves
to both IPv4 and IPv6. While the convert_client only accepts a single
converter address via the CONVERT_ADDR env variable, curl/wget will be
doing happy eyeball and will thus necessarily open a socket with an
IP family different from the converter address, hence crashing the
client.

Signed-off-by: Gregory Vander Schueren <gregory.vanderschueren@tessares.net>

### CODE REVIEW

**Comment by GregoryVds** [comment]
@nkeukeleire Could you test this with google.com? 
Unfortunately I have Ubuntu 20 on my machine, which is not compatible with libsyscall_intercept

File: convert_client.c [hunk 1]
```diff
@@ -650,29 +558,28 @@ _validate_parameters(char *err_buf, size_t len)
 	}
 
 	/* resolve address */
-	if (_validate_and_set_convert_addr(convert_addr) < 0) {
+	if (_validate_and_set_convert_addr(convert_addr) != 0) {
```
Comment (gdetal) [inline convert_client.c]: can be replaced by `getaddrinfo(convert_addr, _convert_port, NULL, &_converter_addr)` directly

**Review by gdetal — COMMENTED** [review]
LGTM, I only have a small comment. This simplifies greatly the code :+1:

File: convert_client.c (hunk 1 above)
Comment (GregoryVds) [reply convert_client.c]: Just fixed that, thanks

**Pull request merged**